Add Loading Indicators to Auth Screen Buttons

tse00022 opened the pull request:
Description:
## Changes Made
- Added loading spinners to Login and Create account buttons
- Enhanced user feedback during authentication processes
- Disabled buttons during loading states to prevent multiple submissions

## Demo
![Demo](https://github.com/user-attachments/assets/989051a3-9bea-422e-a31f-899010a5a1fc)

tse00022 committed:
Add loading indicator to AuthScreen

vladimircezar commented on RedPill/Modules/Onboarding/AuthScreen.swift:
```diff
@@ -32,11 +33,29 @@ struct AuthScreen: View {
+                        ProgressView() // Added spinner
+                            .progressViewStyle(CircularProgressViewStyle(tint: .white))
+                            .scaleEffect(0.8)
+                    }
```
These lines (50 to 54) are exactly the same as 38 to 42. Can we extract this into a view? Maybe we could call it CustomLoadingView (or anything you prefer) and use it there instead. This way, if we ever need to change it we could change them in ne single place.

tse00022 replied:
Good suggestion, we can do it

tse00022 committed:
extract loading button into reusable component

tse00022 commented:
Thank you for the code review feedback! 

I've made the following improvements:
- Extracted the duplicated loading button code into a new reusable `LoadingButton` component
- Replaced both instances in AuthScreen with the new component
- This change improves maintainability and makes future updates more consistent

The code is now more DRY and any future modifications to the loading button UI will only need to be made in one place.

Please let me know if you'd like to see any additional changes.
